### PR TITLE
retry later when sync to clusters failed

### PR DESCRIPTION
**Pull request by ansinlee**
**What this PR does / why we need it**:
when sync to cluster failed, we need wait for the next federated resource changed for the next reconcile, it is indeterminacy, so we need retry later 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by RainbowMango** [comment]
Hi @ansinlee just a remember, your commit `cb92531` may need be squashed.

**Comment by hectorj2f** [comment]
@ansinlee please, squash the commit into a single one, thanks

**Comment by hectorj2f** [comment]
ideally it'd prefer to have more details about this issue, and a test to understand the problem and solution.

File: pkg/controller/sync/controller.go [hunk 1]
```diff
@@ -378,7 +378,11 @@ func (s *KubeFedSyncController) syncToClusters(fedResource FederatedResource) ut
 	}
 
 	collectedStatus := dispatcher.CollectedStatus()
-	return s.setFederatedStatus(fedResource, status.AggregateSuccess, &collectedStatus)
+	status := s.setFederatedStatus(fedResource, status.AggregateSuccess, &collectedStatus)
+	if !ok {
+		return util.StatusNeedsRecheck
+	}
```
Comment (makkes) [inline pkg/controller/sync/controller.go:384]: It is preferable to make use of the `ok` variable right after it has been assigned, i.e. on line 370. This way it's much easier to follow the reasoning of why it is assigned in the first place.

**Review by makkes — CHANGES_REQUESTED** [review]

**Comment by makkes** [comment]
@ansinlee are you still working on this? Asking because there has not been activity on it for quite some time now. 🙂

**Pull request closed without merging**